src/VisualStudio/Core/Def/Implementation/Utilities/NavigationHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Utilities
+{
+    internal static class NavigationHelper
+    {
+        public static bool TryNavigateToPosition(this IServiceProvider serviceProvider, string filePath, int line, int column)
+        {
+            var docTable = (IVsRunningDocumentTable)serviceProvider.GetService(typeof(SVsRunningDocumentTable));
+            var textManager = (IVsTextManager)serviceProvider.GetService(typeof(SVsTextManager));
+            if (docTable == null || textManager == null)
+            {
+                return false;
+            }
+
+            return TryNavigateToPosition(docTable, textManager, filePath, line, column);
+        }
+
+        public static bool TryNavigateToPosition(this VisualStudioWorkspaceImpl workspace, string filePath, int line, int column)
+        {
+            var docTable = workspace.GetVsService<SVsRunningDocumentTable, IVsRunningDocumentTable>();
+            var textManager = workspace.GetVsService<SVsTextManager, IVsTextManager>();
```

Comment:
I've yanked the GetVsService helper from VisualStudioWorkspace in #14354. Please just get an IServiceProvider directly to avoid more coupling to VisualStudioWorkspaceImpl then we already need.